lib/core/versions/latest/DocumentComposer.ts

```diff
@@ -24,10 +24,11 @@ export default class DocumentComposer {
 
     const document = didState.document as DocumentModel;
 
-    // Only populate `publicKey` if general purpose exists.
-    // Only populate `authentication` if authentication purpose exists.
-    const authentication: any[] = [];
-    const publicKeys: any[] = [];
+    // Put each public key in verificationMethod
+    // then populate the verification relationships by reference if a key has multiple purposes,
+    // or populate the verification relattionship by full description if a key has only 1 purpose
```

Comment:
Sanity: I thought Daniel has confirm that we do by reference always?

Reply:
Old comments before we made the decision, will update.